packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -132,28 +161,41 @@ export class DefaultChannelSupport implements ChannelSupport {
       },
     };
     await this.messaging.exchange<JoinUserChannelResponse>(request, 'joinUserChannelResponse');
-
-    this.channelSelector.updateChannel(id, this.userChannels ?? []);
-
-    for (const l of this.followingListeners) {
+    this.channelSelector.updateChannel(id, this.userChannels);
+    for (const l of this.userChannelListeners) {
       await l.changeChannel(new DefaultChannel(this.messaging, id, 'user'));
```

Comment:
It looks like there are 2 separate patterns for changing all the channels. 
Line 152: `this.userChannelListeners.forEach(l => l.changeChannel(null));`
Line 166:
```
for (const l of this.userChannelListeners) {
  await l.changeChannel(new DefaultChannel(this.messaging, id, 'user'));
}
```

Should they both use the same pattern?

Reply:
Yes they should, fixing that on the for loop style (as I find it slightly easier to read than forEach).